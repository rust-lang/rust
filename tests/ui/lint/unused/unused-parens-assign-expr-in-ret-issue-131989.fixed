//@ run-rustfix
#![deny(unused_parens)]
#![allow(unreachable_code)]

fn foo() {
    loop {
        break (_ = 42);
        // lint unused_parens should not be triggered here.
    }

    let _ = loop {
        let a = 1;
        let b = 2;
        break a + b; //~ERROR unnecessary parentheses
    };

    loop {
        if break return () {
            //~^ ERROR unnecessary parentheses
        }
        if break return () {
            //~^ ERROR unnecessary parentheses
        }
    }

    return (_ = 42);
    // lint unused_parens should not be triggered here.
}

fn main() {
    let _ = foo();
}
