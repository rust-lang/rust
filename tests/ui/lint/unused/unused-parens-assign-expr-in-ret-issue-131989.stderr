error: unnecessary parentheses around `break` value
  --> $DIR/unused-parens-assign-expr-in-ret-issue-131989.rs:14:15
   |
LL |         break (a + b);
   |               ^     ^
   |
note: the lint level is defined here
  --> $DIR/unused-parens-assign-expr-in-ret-issue-131989.rs:2:9
   |
LL | #![deny(unused_parens)]
   |         ^^^^^^^^^^^^^
help: remove these parentheses
   |
LL -         break (a + b);
LL +         break a + b;
   |

error: unnecessary parentheses around `if` condition
  --> $DIR/unused-parens-assign-expr-in-ret-issue-131989.rs:18:12
   |
LL |         if (break return ()) {
   |            ^               ^
   |
help: remove these parentheses
   |
LL -         if (break return ()) {
LL +         if break return () {
   |

error: unnecessary parentheses around `break` value
  --> $DIR/unused-parens-assign-expr-in-ret-issue-131989.rs:21:18
   |
LL |         if break (return ()) {
   |                  ^         ^
   |
help: remove these parentheses
   |
LL -         if break (return ()) {
LL +         if break return () {
   |

error: aborting due to 3 previous errors

